### ISSUE TITLE
Revisit the use of references to crossplane-system

### ISSUE THREAD

**Issue by bassam** [crossplaneio/crossplane#89]
We currently use a full `ObjectReference` on abstract resources to identify the `ResourceClass` to be used. For example:

```yaml
apiVersion: storage.conductor.io/v1alpha1
kind: MySQLInstance
...
spec:
  classReference:
    name: standard
    namespace: conductor-system
```

In this approach a developer would need to know the name of the class `standard` and the namespace in which conductor operator is running `conductor-system`. Once approach is simplify the classReference and just use a `className` and assume that the namespace is wherever the conductor operator is running.

```yaml
apiVersion: storage.conductor.io/v1alpha1
kind: MySQLInstance
...
spec:
  className: standard
```

This approach is closer to how `StorageClass` is identified in PVC design. Storage classes are cluster scope while `ResourceClass` is at a namespace scope.

This also opens up the whole discussion on how an application developer finds out about available resources classes. Do they `kubectl get rc -n conductor-system`? do they have access to list classes in that namespace? Can they look at the spec of the resource class or would it contain sensitive information?

**Comment by prasek** [comment]
@jbw976 @hasheddan should this be closed with #151?

**Comment by hasheddan** [comment]
@prasek I don't believe so. From what I understand, the discussion of making resource classes cluster-scoped is still open, and the strongly typed resource classes can currently be created in any namespace.

**Comment by prasek** [comment]
thanks @hasheddan, now that I read this more closely it’s not about removing the classReference but simplifying it to assume a singular system namespace (cluster-scoped or crossplane-system). Given that would this be closed as part of #609 / #663? 

//cc @jbw976

**Comment by hasheddan** [comment]
So yes, this should be closed by #663, in my opinion. However, I think it is important to note that this issue is a very specific question around how referencing of resource classes is currently accomplished. Even though all resource classes currently are namespace-scoped, if cluster-scoped resource classes were introduced, they could still be referenced in the same manner. The `classRef` field is just a Kubernetes `ObjectReference` and can refer to an object that is either namespace-scoped *or* cluster-scoped. I imagine we will reach a point (as informed by the language in #609 / #663) where we have resource classes that fall in both categories. We should have no problem referencing either type. Thanks for pointing this out!

**Comment by negz** [comment]
I think this issue can be closed given the changes introduced in 0.3. Specifically

* We're no longer advising `crossplane-system` as a namespace for anything other than running  Crossplane and infra stack pods. We recommend using namespaces of your own design to logically group "infrastructure stuff" like providers, non-portable resource classes, and managed resources.
* Claims now reference a non-portable resource class via a portable resource class. Claims use a LocalObjectReference to their non-portable resource class, which must be in the namespace of the claim. The non-portable resource class uses an object reference to a class of an arbitrary kind in an arbitrary namespace.

Please reopen if there's still work to track here.
